StoreKitUnitTests/TrialOrIntroPriceEligibilityCheckerTests.swift

```diff
@@ -0,0 +1,189 @@
+        }
+
+        expect(self.receiptFetcher.receiptDataCalled).to(beTrue())
+    }
+
+    func testSK1EligibilityIsCalculatedFromReceiptData() throws {
+        setupSK1()
+        let stubbedEligibility = ["product_id": IntroEligibilityStatus.eligible]
+        mockIntroEligibilityCalculator.stubbedCheckTrialOrIntroductoryPriceEligibilityResult = (stubbedEligibility, nil)
+
+        var completionCalled = false
+        var maybeEligibilities: [String: IntroEligibility]?
+        trialOrIntroPriceEligibilityChecker!.sk1CheckEligibility([]) { (eligibilities) in
+            completionCalled = true
+            maybeEligibilities = eligibilities
+        }
+
+        expect(completionCalled).toEventually(beTrue())
+        let receivedEligibilities = try XCTUnwrap(maybeEligibilities)
+        expect(receivedEligibilities.count) == 1
+    }
+
+    func testSK1EligibilityIsFetchedFromBackendIfErrorCalculatingEligibility() throws {
+        setupSK1()
+        let stubbedError = NSError(domain: RCPurchasesErrorCodeDomain,
+                                   code: ErrorCode.invalidAppUserIdError.rawValue,
+                                   userInfo: [:])
+        mockIntroEligibilityCalculator.stubbedCheckTrialOrIntroductoryPriceEligibilityResult = ([:], stubbedError)
+
+        let stubbedEligibility = ["product_id": IntroEligibility(eligibilityStatus: IntroEligibilityStatus.eligible)]
+        mockBackend.stubbedGetIntroEligibilityCompletionResult = (stubbedEligibility, nil)
+        var completionCalled = false
+        var maybeEligibilities: [String: IntroEligibility]?
+        trialOrIntroPriceEligibilityChecker!.sk1CheckEligibility([]) { (eligibilities) in
+            completionCalled = true
+            maybeEligibilities = eligibilities
+        }
+
+        expect(completionCalled).toEventually(beTrue())
+        let receivedEligibilities = try XCTUnwrap(maybeEligibilities)
+        expect(receivedEligibilities.count) == 1
+    }
+
+    func testSK1ErrorFetchingFromBackendAfterErrorCalculatingEligibility() throws {
+        setupSK1()
+        let stubbedError = NSError(domain: RCPurchasesErrorCodeDomain,
+                                   code: ErrorCode.invalidAppUserIdError.rawValue,
+                                   userInfo: [:])
+        mockIntroEligibilityCalculator.stubbedCheckTrialOrIntroductoryPriceEligibilityResult = ([:], stubbedError)
+
+        mockBackend.stubbedGetIntroEligibilityCompletionResult = ([:], stubbedError)
+        var completionCalled = false
+        var maybeEligibilities: [String: IntroEligibility]?
+        trialOrIntroPriceEligibilityChecker!.sk1CheckEligibility([]) { (eligibilities) in
+            completionCalled = true
+            maybeEligibilities = eligibilities
+        }
+
+        expect(completionCalled).toEventually(beTrue())
+        let receivedEligibilities = try XCTUnwrap(maybeEligibilities)
+        expect(receivedEligibilities.count) == 0
```

Comment:
should this be returning unknown?

Reply:
I think so, that's what would happen with SK2. Will make the change